File: Android/app/src/main/java/com/apps/kruszyn/lightorganapp/ui/CircleView.java
package com.apps.kruszyn.lightorganapp.ui;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.view.View;

import com.apps.kruszyn.lightorganapp.R;

/**
 * TODO: document your custom view class.
 */
public class CircleView extends View {

    private int mCircleColor = Color.RED;

    private Paint mPaint;

    public CircleView(Context context) {
        super(context);
        init(null, 0);
    }

    public CircleView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(attrs, 0);
    }

    public CircleView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init(attrs, defStyle);
    }

    private void init(AttributeSet attrs, int defStyle) {
        // Load attributes
        final TypedArray a = getContext().obtainStyledAttributes(
                attrs, R.styleable.CircleView, defStyle, 0);

        mCircleColor = a.getColor(
                R.styleable.CircleView_circleColor,
                mCircleColor);

        a.recycle();

        mPaint = new Paint();
        mPaint.setStyle(Paint.Style.FILL);
        mPaint.setColor(mCircleColor);
    }

    private void invalidatePaint() {
        mPaint.setColor(mCircleColor);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);

        // TODO: consider storing these as member variables to reduce
        // allocations per draw cycle.
        int paddingLeft = getPaddingLeft();
        int paddingTop = getPaddingTop();
        int paddingRight = getPaddingRight();
        int paddingBottom = getPaddingBottom();

        int contentWidth = getWidth() - paddingLeft - paddingRight;
        int contentHeight = getHeight() - paddingTop - paddingBottom;
        float radius = Math.min(contentWidth, contentHeight) / 2;

        canvas.drawCircle(paddingLeft + contentWidth / 2,
                paddingTop + contentHeight / 2,
                radius,
                mPaint);
    }


    public int getCircleColor() {
        return mCircleColor;
    }

    public void setCircleColor(int circleColor) {
        mCircleColor = circleColor;
        invalidatePaint();
    }
}

File: Xamarin/LightOrganApp.Droid/UI/CircleView.cs

using Android.Views;
using Android.Graphics;
using Android.Util;
using Android.Content;
using System;

namespace LightOrganApp.Droid.UI
{
    public class CircleView: View
    {
        private Color circleColor = Color.Red;

        private Paint paint;

        public CircleView(Context context): base(context)
        {            
            Init(null, 0);
        }

        public CircleView(Context context, IAttributeSet attrs): base(context, attrs)
        {
            Init(attrs, 0);
        }

        public CircleView(Context context, IAttributeSet attrs, int defStyle): base(context, attrs, defStyle)
        {
            Init(attrs, defStyle);
        }

        public Color CircleColor
        {
            get
            {
                return circleColor;
            }

            set
            {
                circleColor = value;
                InvalidatePaint();
            }
        }

        private void Init(IAttributeSet attrs, int defStyle)
        {
            //Load attributes
            var a = Context.ObtainStyledAttributes(
                    attrs, Resource.Styleable.CircleView, defStyle, 0);

            circleColor = a.GetColor(
                    Resource.Styleable.CircleView_circleColor,
                    circleColor);

            a.Recycle();

            paint = new Paint();
            paint.SetStyle(Paint.Style.Fill);
            paint.Color = circleColor;
        }

        private void InvalidatePaint()
        {
            paint.Color = circleColor;
            Invalidate();
        }
       
        protected override void OnDraw(Canvas canvas)
        {
            base.OnDraw(canvas);

            // TODO: consider storing these as member variables to reduce
            // allocations per draw cycle.
            int paddingLeft = PaddingLeft;
            int paddingTop = PaddingTop;
            int paddingRight = PaddingRight;
            int paddingBottom = PaddingBottom;

            int contentWidth = Width - paddingLeft - paddingRight;
            int contentHeight = Height - paddingTop - paddingBottom;
            float radius = Math.Min(contentWidth, contentHeight) / 2;

            canvas.DrawCircle(paddingLeft + contentWidth / 2,
                    paddingTop + contentHeight / 2,
                    radius,
                    paint);
        }
    }   
}